src/addons/grids/ReactDataGrid.js

```diff
@@ -86,31 +86,46 @@ const ReactDataGrid = React.createClass({
     return initialState;
   },
 
-  onSelect: function(selected: SelectedType) {
-    if (this.props.enableCellSelect) {
-      if (this.state.selected.rowIdx !== selected.rowIdx
-        || this.state.selected.idx !== selected.idx
-        || this.state.selected.active === false) {
-        let idx = selected.idx;
-        let rowIdx = selected.rowIdx;
-        if (
-            idx >= 0
-            && rowIdx >= 0
-            && idx < ColumnUtils.getSize(this.state.columnMetrics.columns)
-            && rowIdx < this.props.rowsCount
-          ) {
-          this.setState({selected: selected});
-        }
+  hasSelectedCellChanged: function(selected: SelectedType) {
+    let previouslySelected = Object.assign({}, this.state.selected);
+    return previouslySelected.rowIdx !== selected.rowIdx || previouslySelected.idx !== selected.idx || previouslySelected.active === false;
+  },
+
+  onSelect: function(selected: SelectedType, associatedEvent: string) {
```

Comment:
To implement a generic behaviour and don't violate SRP associated events should not be bound with `onSelect`. 
The problem comes when the state needs to change, how can we make sure outside the `setState` callback that when `onColumnEvent `is triggered the state will contain the last selected column/Cell